src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -704,7 +720,14 @@ public static void CopyArray (IntPtr src, string[] dest)
 					AssertIsJavaObject (type);
 
 					IntPtr elem = GetObjectArrayElement (source, index);
-					return Java.Lang.Object.GetObject (elem, JniHandleOwnership.TransferLocalRef, type);
+					return GetObject (elem, type);
+
+					// FIXME: https://github.com/xamarin/xamarin-android/issues/8724
+					// Since a Dictionary<Type, Func> is used here, the trimmer will not be able to properly analyze `Type t`
+					// error IL2111: Method 'lambda expression' with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
+					[UnconditionalSuppressMessage ("Trimming", "IL2067", Justification = "[DynamicallyAccessedMembers] manually added to all callers")]
+					static object? GetObject (IntPtr e, Type t) =>
+						Java.Lang.Object.GetObject (e, JniHandleOwnership.TransferLocalRef, t);
```

Comment:
I didn't find a way to fix this properly yet. I think the pattern of:

* Create a `Dictionary<Type, Func>` where one of the parameters needs `[DynamicallyAccessedMembers]`
* If you decorate the lambda, or create a method...
* The trimmer says: "Method 'lambda expression' with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method."

This situation just isn't going to make the trimmer happy.

For now, I linked to:

* https://github.com/xamarin/xamarin-android/issues/8724

The future fix might be to remove the `Dictionary` in favor of `switch`, that seems like the trimmer could more easily understand.